### PR TITLE
Online DDL: ddl_strategy=direct

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Backport
NO

## Status

ready

## Description

Following an internal discussion, where @askdba noticed this output:
```
 vtctlclient ApplySchema -sql="$(cat create_commerce_schema.sql)" commerce
Handling connection for 15999
I1213 17:07:05.817487   35734 main.go:64] I1213 14:07:05.661886 tablet_executor.go:259] Received DDL request. strategy =
I1213 17:07:05.882329   35734 main.go:64] I1213 14:07:05.729416 tablet_executor.go:259] Received DDL request. strategy =
```

`strategy = ` (empty) seems confusing. Up till now, the empty string stood for the "direct" (non-online) DDL strategy.

Starting this PR, the dircet (non-online) strategy is named `"direct"`. Consider the following examples:

```sql
mysql> select @@ddl_strategy;
+----------------+
| @@ddl_strategy |
+----------------+
| direct         |
+----------------+
1 row in set (0.00 sec)

mysql> alter table corder modify order_id int;
Query OK, 0 rows affected (0.03 sec)

mysql> set @@ddl_strategy='gh-ost';
Query OK, 0 rows affected (0.00 sec)

mysql> alter table corder modify order_id int;
+--------------------------------------+
| uuid                                 |
+--------------------------------------+
| 1305a8e4_3dd5_11eb_85ea_f875a4d24e90 |
+--------------------------------------+
1 row in set (0.00 sec)

mysql> set @@ddl_strategy='';
Query OK, 0 rows affected (0.00 sec)

mysql> alter table corder modify order_id int;
Query OK, 0 rows affected (0.01 sec)
```

```shell
/tmp/local$ vtctlclient ApplySchema -sql "alter table corder modify order_id int" commerce
I1214 08:25:23.239231 1630821 main.go:64] I1214 06:25:23.238050 tablet_executor.go:272] Received DDL request. strategy=direct
/tmp/local$ vtctlclient ApplySchema -ddl_strategy "gh-ost" -sql "alter table corder modify order_id int" commerce
I1214 08:25:38.917898 1630847 main.go:64] I1214 06:25:38.917187 tablet_executor.go:272] Received DDL request. strategy=gh-ost
I1214 08:25:38.918173 1630847 main.go:64] I1214 06:25:38.917839 tablet_executor.go:308] UUID=2ee751f1_3dd5_11eb_a6e5_f875a4d24e90
2ee751f1_3dd5_11eb_a6e5_f875a4d24e90
/tmp/local$ vtctlclient ApplySchema -ddl_strategy "" -sql "alter table corder modify order_id int" commerce
I1214 08:25:44.846210 1630860 main.go:64] I1214 06:25:44.844765 tablet_executor.go:272] Received DDL request. strategy=direct
```

notes:
- in vtgate, the default value for `-ddl_strategy` command line flag is `direct`. An empty value is also interpreted as `direct`.
- likewise, in vtctl/vtctlclient, the default value for `-ddl_strategy` command line flag is `direct`. An empty value is also interpreted as `direct`.

## Related Issue(s)

Tracking: #6926 

## Todos
- [x] Tests
- [x] Documentation

## Impacted Areas in Vitess
List general components of the application that this PR will affect:

- [x]  Query Serving
- [ ]  VReplication
- [x]  Cluster Management
- [ ]  Build

### CODE REVIEW

**Comment by shlomi-noach** [comment]
Documentation in https://github.com/vitessio/website/pull/639

**Comment by askdba** [comment]
Did we intentionally used dd_strategy instead of ddl_strategy? 
```in vtgate, the default value for -dd_strategy command line flag is direct. An empty value is also interpreted as direct.
likewise, in vtctl/vtctlclient, the default value for -dd_strategy command line flag is direct. An empty value is also interpreted as direct.```

**Comment by shlomi-noach** [comment]
> Did we intentionally used dd_strategy instead of ddl_strategy?

That's just a typo in my comment. Fixed.

**Comment by askdba** [comment]
On the MySQL CLI I've got the same results as above test case.
```MySQL [commerce]> set @@ddl_strategy='gh-ost';
Query OK, 0 rows affected (0.00 sec)

MySQL [commerce]> alter table corder modify order_id int;
+--------------------------------------+
| uuid                                 |
+--------------------------------------+
| fc1d1705_404a_11eb_b95d_0242ac110002 |
+--------------------------------------+
1 row in set (0.00 sec)

MySQL [commerce]>
MySQL [commerce]>
MySQL [commerce]> set @@ddl_strategy='direct';
Query OK, 0 rows affected (0.00 sec)

MySQL [commerce]> alter table corder modify order_id int;
Query OK, 0 rows affected (0.01 sec)
```
But on the command line, I did not get the same output. 
```
vitess@5c644a6f4170:/vt/local$ vtctlclient ApplySchema -sql "alter table corder modify order_id bigint" commerce
vitess@5c644a6f4170:/vt/local$ vtctl --version
ERROR: logging before flag.Parse: E1217 09:36:03.895258    1618 syslogger.go:149] can't connect to syslog
Version: d65877152 (Git branch 'ddl-strategy-direct') built on Thu Dec 17 09:15:19 UTC 2020 by vitess@ecd12ae46f97 using go1.13.15 linux/amd64
```

**Comment by shlomi-noach** [comment]
> But on the command line, I did not get the same output.

That's because since this PR was opened, I already merged another PR that removes that log line altogether. If you'd like the log line to appear please let me know.

**Comment by askdba** [comment]
I've found the log message useful. I'd like it back.

**Comment by shlomi-noach** [comment]
Log message is now restored.

**Comment by shlomi-noach** [comment]
:pray: request for additional review

**Review by ajm188 — APPROVED** [review]
lgtm

File: go/vt/schema/online_ddl.go [hunk 1]
```diff
@@ -79,14 +79,24 @@ const (
...
+	// DDLStrategyDirect means not an online-ddl migration. Just a normal MySQL ALTER TABLE
+	DDLStrategyDirect DDLStrategy = "direct"
 	// DDLStrategyGhost requests gh-ost to run the migration
 	DDLStrategyGhost DDLStrategy = "gh-ost"
 	// DDLStrategyPTOSC requests pt-online-schema-change to run the migration
 	DDLStrategyPTOSC DDLStrategy = "pt-osc"
 )
 
+// IsDirect returns true if this strategy is a direct strategy
+// A strategy is direct if it's not explciitly one of the online DDL strategies
```
Comment (indera-shsp) [inline go/vt/schema/online_ddl.go:91]: nit pick: `explciitly`

File: go/vt/schemamanager/tablet_executor.go [hunk 2]
```diff
@@ -232,6 +232,7 @@ func (exec *TabletExecutor) executeSQL(ctx context.Context, sql string, execResu
 			return nil
 		}
 	}
+	exec.wr.Logger().Infof("Received DDL request. strategy=%+v", schema.DDLStrategyDirect)
```
Comment (indera-shsp) [inline go/vt/schemamanager/tablet_executor.go:235]: Perhaps it is useful to log the `sql` too 🤷‍♂️

**Pull request merged**